### PR TITLE
Feature/583 circular hierarchy

### PR DESCRIPTION
**Pull request by PROgram52bc**
- Utility functions and tests related to resolving the group leadership hierarchy
- Checks for circular structure when:
   - adding a manager/member
   - reactivating a manager/member
   - moving a manager/member
   - reactivating a group (not implemented)
- fixed some/most of the styling/linting errors in the group module

### CODE REVIEW

File: ui/src/components/groups/GroupParticipants.vue [hunk 1]
```diff
@@ -582,7 +580,31 @@ export default {
           participant.disabledText = this.$t(
             "groups.batch-actions.messages.person-in-destination-group"
           );
-        } // else if (moving participant will create cycle) {...}
+        } else {
+          // check whether moving moving participant will create cycle
```
Comment (PROgram52bc) [inline ui/src/components/groups/GroupParticipants.vue]: typo

File: ui/src/models/GroupHierarchyNode.ts [hunk 2]
```diff
@@ -0,0 +1,488 @@
...
+      parentNode.children.push(childNode);
+    }
+  );
+  return infoRootNode;
+}
+
+/* checks whether a node is a root node
+ * a root node <r> is a node that either:
+ * 1. does not have any super-node, or
+ * 2. for all its immediate super-node <s> such that <r> is the only super-node of <s>
```
Comment (PROgram52bc) [inline ui/src/models/GroupHierarchyNode.ts]: rephrase this

**Comment by nurkkala** [comment]
This is a big one, so I would rather review it when there are no conflicts.

Please resolve merge conflicts and resubmit.

**Comment by PROgram52bc** [comment]
I will resolve it soon!

File: ui/src/components/groups/treeview/GroupTreeViewHierarchy.vue [hunk 3]
```diff
@@ -0,0 +1,205 @@
+<template>
+  <div>
+    <v-card v-if="cycleError.show" flat>
+      <v-card-title class="justify-center">
+        <v-alert color="warning">
+          <v-tooltip bottom>
+            <template v-slot:activator="{ on }">
+              <v-icon v-on="on">error</v-icon>
+            </template>
+            Please remove the cycle in the node listed in order to view group
```
Comment (PROgram52bc) [inline ui/src/components/groups/treeview/GroupTreeViewHierarchy.vue:10]: internationalize this

**Comment by PROgram52bc** [comment]
merged into #601

**Comment by PROgram52bc** [comment]
merged into #601

**Pull request closed without merging**